src/Storages/MergeTree/MergeTreeSettings.h

```diff
@@ -186,6 +186,8 @@ struct Settings;
     M(Bool, disable_freeze_partition_for_zero_copy_replication, true, "Disable FREEZE PARTITION query for zero copy replication.", 0) \
     M(Bool, disable_detach_partition_for_zero_copy_replication, true, "Disable DETACH PARTITION query for zero copy replication.", 0) \
     M(Bool, disable_fetch_partition_for_zero_copy_replication, true, "Disable FETCH PARTITION query for zero copy replication.", 0) \
+    M(Bool, enable_block_number_column, false, "Enable persisting column _block_number for each row.", 0) ALIAS(allow_experimental_block_number_column) \
```

Comment:
If these settings are not experimental anymore we should randomize them in CI. The last time that detected a number of issues IIRC and we had to remove it https://github.com/ClickHouse/ClickHouse/pull/56474